lib/factory_bot/sequence.rb

```diff
@@ -55,15 +55,33 @@ def initialize(value)
       end
 
       def peek
-        @value
+        value
       end
 
       def next
-        @value = @value.next
+        @value = value.next
       end
 
       def rewind
-        @value = @first_value
+        @value = first_value
+      end
+
+      private
+
+      def value
+        if @value.respond_to?(:call)
+          @value = @value.call
+        else
+          @value
+        end
+      end
+
+      def first_value
```

Comment:
Calling the block again when we `rewind` means that rewinding will not bring us back to the value we started at. This sort of goes against why we introduced the feature to rewind sequences - it was for people who were relying on specific ids in their tests for one reason or another.

If we go forward with this feature, I think we will need some additional test coverage around how this works with rewinding sequences.